lib/actions/build.coffee

```diff
@@ -0,0 +1,224 @@
+		pack.finalize()
+		return pack
+
+getAppInfo = (application) ->
+	resin = require('resin-sdk-preconfigured')
+	return resin.models.application.get(application)
+	.then (app) ->
+		if !app?
+			throw new Error('Could not read application information!')
+		Promise.all([app, resin.models.config.getDeviceTypes()])
+	.then ([app, config]) ->
+		config = _.find(config, 'slug': app.device_type)
+		if !config?
+			throw new Error('Could not read application information!')
+		app.arch = config.arch
+		return app
+
+getBundleInfo = (options) ->
+	if options.application?
+		# An application was provided
+		getAppInfo(options.application)
+		.then (app) ->
+			console.log('Building for application: ' + app.app_name)
+			return [app.arch, app.device_type]
+		.catch (err) ->
+			throw err
+	else if options.arch? and options.deviceType?
+		return Promise.resolve([options.arch, options.deviceType])
+	else
+		# No information, cannot do resolution
+		return
+
+generateConnectOpts = (options) ->
+	connectOpts = {}
+	# Firsly need to decide between a local docker socket
+	# and a host available over a host:port combo
+	if options.docker? and not options.dockerHost?
+		# good, local docker socket
+		connectOpts['socketPath'] = options.docker
+	else if options.dockerHost? and not options.docker?
+		# Good a host is provided, and local socket isn't
+		connectOpts['host'] = options.dockerHost
+		connectOpts['port'] = options.dockerPort || 2376
+	else if options.docker? and options.dockerHost?
+		# Both provided, no obvious way to continue
+		throw new Error("Both a local docker socket and docker host have been provided. Don't know how to continue.")
+	else
+		# None provided, assume default docker local socket
+		connectOpts['socketPath'] = '/var/run/docker.sock'
+
+	# Now need to check if the user wants to connect over TLS
+	# to the host
+	connectOpts['ca'] = options.ca if options.ca?
+	connectOpts['cert'] = options.cert if options.cert?
+	connectOpts['key'] = options.key if options.key?
+
+	return connectOpts
+
+module.exports =
+	signature: 'build [context]'
+	description: 'Build a container locally'
+	permission: 'user'
+	options: [
+		{
+			signature: 'docker'
+			parameter: 'docker'
+			description: 'Path to a local docker socket'
+			alias: 'P'
+		},
+		{
+			signature: 'arch'
+			parameter: 'arch'
+			description: 'The architecture to build for'
+			alias: 'A'
+		},
+		{
+			signature: 'deviceType'
+			parameter: 'deviceType'
+			description: 'The type of device this build is for'
+			alias: 'd'
+		},
+		{
+			signature: 'context'
+			parameter: 'context'
+			description: "The build context, which contains the build files. Defaults to '.'"
+			alias: 'c'
+		},
+		{
+			signature: 'application'
+			parameter: 'application'
+			description: 'The target resin.io application this build is for'
+			alias: 'a'
+		},
+		{
+			signature: 'tag'
+			parameter: 'tag'
+			description: 'The alias to the generated image'
+			alias: 't'
+		},
+		{
+			signature: 'nocache'
```

Comment:
It's because `nocache` is the docker standard way of passing it, but I've deviated away from how docker names other things already so I might as well keep my inconsistencies consistent :p